### ISSUE TITLE
Some QuantumCircuit methods do not pass label and ctrl_state for controlled gates

### ISSUE THREAD

**Issue by enavarro51** [Qiskit/qiskit-terra#5098]
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.8
- **Operating system**: Ubuntu 18.04

### What is the current behavior?
The following controlled-gate methods in QuantumCircuit do not accept label or ctrl_state as a param.
- ccx, toffoli, mcx, rccx, rcccx, fredkin, mcu1
### Steps to reproduce the problem
```
from qiskit import QuantumCircuit

qc = QuantumCircuit(3)
qc.cx(0, 1, label='Label', ctrl_state=0)
qc.ccx(0, 1, 2, label='Label', ctrl_state=1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-decb3879596f> in <module>
      3 qc = QuantumCircuit(3)
      4 qc.cx(0, 1, label='Label', ctrl_state=0)
----> 5 qc.ccx(0, 1, 2, label='Label', ctrl_state=1)

TypeError: ccx() got an unexpected keyword argument 'label'
```
### What is the expected behavior?
All the above gates should accept label and ctrl_state
### Suggested solutions
Add label and ctrl_state to the above methods in QuantumCircuit class and pass them through to gate classes.

**Comment by shifubear** [comment]
Definitely seems odd, especially since the controlled-gate classes for the ones without labels are subclasses of the same ControlledGate class, with an initializer taking labels as parameters. I'd be happy to work on this!

**Comment by 1ucian0** [comment]
I agree this is an inconsistency. However, I think the solution is to remove them from those that have the `label`, not adding to those that are missing it. If a label is required, I think is reasonable to use the `.append` notation.

**Comment by 1ucian0** [comment]
After some internal discussion it has been decided that the most appropriate fix this issue is to remove the label parameter from the QuantumCircuit methods that have it. I'm leaving this issue for the `label` part of it.

About `ctrl_state`, moved to https://github.com/Qiskit/qiskit-terra/issues/6465

**Comment by eptru** [comment]
Newcomer here. I am interested in taking this once we reach a decision! I can take the label issue at the same time.

**Comment by javabster** [comment]
Thanks @eptru! it looks like #6465 is already being worked on, but I'll assign this one to you 😄 Let us know if you have any questions!

**Comment by fs1132429** [comment]
I am new to contributing and I would love to work on this if its ok.

**Comment by javabster** [comment]
Hi @fs1132429 thanks for your interest but this issue has already been assigned and is being worked on. If you go to the issues tab and search for good first issues there are many that have not yet been assigned to someone which you might like to take on 😄

**Comment by aprew** [comment]
Considering this seems abandoned would I be able to try fixing this instead?

**Comment by aprew** [comment]
> Sure @aprew go ahead! Let us know if you have any questions, and you can take a look at the following resources as well:
> 
>     * [Contributing Guidelines](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md)
> 
>     * [GitHub / CI tips for Qiskit Contributors](https://www.youtube.com/watch?v=-eLeOeF_-5o)

Will do thank you! :)

**Comment by 1ucian0** [comment]
Assignee move to @aprew. Thanks and let us know if you have time to work on this

**Comment by PriyanshuPansari** [comment]
Is this still being worked on? If not I am interested in trying to fix the issue.

**Comment by ghost** [comment]
Hi, sorry yes it is. I'll spare you the excuses but I will start working on it again soon.

**Comment by jade-emperror** [comment]
@aprew is this still being worked on ? if not i would like to give it a shot

**Comment by muhundet** [comment]
Is this issue still being worked on? I would like to give it a shot also

**Comment by 1ucian0** [comment]
hi @muhundet ! you are in! (sorry for the delay)

**Comment by CB2999** [comment]
I'd also like to throw my hat in the ring, if there's still work to be done here.

**Comment by migueltorrescosta** [comment]
Happy to take this ( assuming it isn't being worked on by @muhundet  nor @CB2999 )

**Comment by 1ucian0** [comment]
The issue is still valid in `2.5.0.dev0+59afe0f` (at least for ccx):
```python
from qiskit import QuantumCircuit
qc = QuantumCircuit(3)
qc.ccx(0, 1, 2, label='Label')
```
```
TypeError: QuantumCircuit.ccx() got an unexpected keyword argument 'label'
```

The current decision is add the `label` keyword argument to all the gates as QuantumCircuit methods and to the gate classes.

**Comment by 1ucian0** [comment]
| :exclamation:  From June 3rd to 17th, this issue is participating in the 6th edition of [unitaryHACK](https://unitaryhack.dev/projects/qiskit/?utm_source=qiskit_github)   <sub>[more info...](https://unitaryhack.dev/)</sub> |
|----------------------------------------------|
<p align="center">
  <img src="https://unitary.foundation/images/2026_uhack_logo.png?raw=true"/>
</p>

If you want to participate, please follow these guidelines:
 * Read and follow [the Hacker Guide](https://unitaryhack.dev/hacker-guide/) and [unitaryHACK Rules](https://unitaryhack.dev/rules/)
 * You can submit a PR without being assigned to the issue. No need to post "I'm working on this" or similar.
 * PRs not following the [unitaryHACK Hacker's Guide to Ethical AI Contributions](https://unitaryhack.dev/ai-guide/) **will be closed directly** and repeat offenders **will be banned** from the Qiskit GitHub organization.
 * No need to mention any of the maintainers or ping them directly. If you have any questions or feedback, use the channel [`#unitaryhack-26` in the Unitary Foundation Discord](https://discord.com/channels/764231928676089909/1489401346473066666)

The winners will be announced **on June 24th**.

Happy hacking!

**Comment by 1ucian0** [comment]
Thank you to everyone who has submitted a PR for this issue! We currently have several promising fixes under review, so **please hold off on submitting new PRs** until the Qiskit team posts a follow-up comment here.

We will review the existing submissions in due course. Thank you for your patience!

**Comment by 1ucian0** [comment]
The PR #16373 is currently the closest to be merged.

@DennisWayo, please address the pending review comments and leave a comment in this issue.

**Comment by DennisWayo** [comment]
> The PR #16373 is currently the closest to be merged.
> 
> @DennisWayo, please address the pending review comments and leave a comment in this issue. 

I will.

**Comment by DennisWayo** [comment]
@1ucian0, the comment has been addressed. 
I added a separate release-note entry for the new `RVGate` `label` constructor parameter.

Checks run:
- `python -m unittest -k quantum_circuit_standard_gate_label test.python.circuit.test_circuit_operations`
- `python -m unittest -k rv test.python.circuit.test_gate_definitions`
- `git diff --check`

**Comment by 1ucian0** [comment]
Closing the issue by hand so the bounty can be assigned, as https://github.com/Qiskit/qiskit/pull/16373 is `on hold` (https://github.com/Qiskit/qiskit/pull/16373#pullrequestreview-4543480812)

Congratulation @DennisWayo !

**Comment by 1ucian0** [comment]
re-opening until https://github.com/Qiskit/qiskit/pull/16373 (which is `on hold`) gets merged.
